### PR TITLE
Openshift

### PR DESCRIPTION
**Pull request by yaroslavkasatikov**
- It doesn't work on OpenShift
 fixed by SCC
 - When running helm upgrade, data contained in configmaps are removed
We decided not to proceed any changing here before helm lookup function will be in prod
 - It's not possible to add provisioners from the helm chart
I investigated this question and my reply here is - it's possible. It works fine with postInitHook.
  bootstrap:
    postInitHook: |-
      step ca provisioner add Google --type oidc --ca-config $(step path)/config/ca.json \
      --client-id 650445034027-jsjdrkiskeq9ke99ud2rqkst82ft8uch.apps.googleusercontent.com \
      --client-secret 6Q7lGMua_Oox4nA92QBXYypT \
      --configuration-endpoint https://accounts.google.com/.well-known/openid-configuration \
      --domain smallstep.com --domain gmail.com
 - Some data that are supposed to be secret are stored in configmaps
 All fixed here. Now it works with Secrets

### CODE REVIEW

File: argocd/argoapp-smallstep.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    targetRevision: openshift
+    path: step-certificates
+    helm:
+      values: |
+        ca:
+          bootstrap:
+            # Add script snippets here to be executed after the step ca init has been run
+            postInitHook: |-
+              step ca provisioner add Google --type oidc --ca-config $(step path)/config/ca.json \
+              --client-id 650445034027-jsjdrkiskeq9ke99ud2rqkst82ft8uch.apps.googleusercontent.com \
```
Comment (aelbarkani) [inline argocd/argoapp-smallstep.yaml]: this is secret information and should not be here

File: argocd/argoapp-smallstep.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (aelbarkani) [inline argocd/argoapp-smallstep.yaml]: ArgoCD app should be moved to here : https://github.com/advatys/kubernetes-library/tree/main/tlsaas/tls-cockpit

File: step-certificates/templates/scc-role-binding.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
```
Comment (aelbarkani) [inline step-certificates/templates/scc-role-binding.yaml]: a role binding would be enough I think

File: step-certificates/templates/scc-role.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
Comment (aelbarkani) [inline step-certificates/templates/scc-role.yaml]: a Role would be enough

**Review by aelbarkani — CHANGES_REQUESTED** [review]

File: argocd/argoapp-smallstep.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      - CreateNamespace=true
+  source:
+    repoURL: https://github.com/advatys/smallstep-helm-charts.git
+    targetRevision: openshift
+    path: step-certificates
+    helm:
+      values: |
+        ca:
+          bootstrap:
+            # Add script snippets here to be executed after the step ca init has been run
```
Comment (aelbarkani) [inline argocd/argoapp-smallstep.yaml]: would be possible to add an example with ACME rather than OIDC ?

**Review by aelbarkani — CHANGES_REQUESTED** [review]

File: argocd/argoapp-smallstep.yaml (hunk 1 above)
Comment (yaroslavkasatikov) [reply argocd/argoapp-smallstep.yaml]: it's example from https://smallstep.com/docs/step-ca/configuration#provisioners

Comment (yaroslavkasatikov) [reply argocd/argoapp-smallstep.yaml]: removed

File: argocd/argoapp-smallstep.yaml (hunk 2 above)
Comment (yaroslavkasatikov) [reply argocd/argoapp-smallstep.yaml]: removed

File: step-certificates/templates/scc-role-binding.yaml (hunk 3 above)
Comment (yaroslavkasatikov) [reply step-certificates/templates/scc-role-binding.yaml]: changed

File: step-certificates/templates/scc-role.yaml (hunk 4 above)
Comment (yaroslavkasatikov) [reply step-certificates/templates/scc-role.yaml]: changed

File: argocd/argoapp-smallstep.yaml (hunk 5 above)
Comment (yaroslavkasatikov) [reply argocd/argoapp-smallstep.yaml]: will be done in another repo. Now added ACME installation by default into values.yaml

**Pull request merged**